nucypher/network/server.py

```diff
@@ -250,6 +257,33 @@ def set_policy(self, id_as_hex, request: Request):
         # TODO: Sign the arrangement here.  #495
         return  # TODO: Return A 200, with whatever policy metadata.
 
+    def revoke_arrangement(self, id_as_hex, request: Request):
+        """
+        REST endpoint for revoking/deleting a KFrag from a node.
+        """
+        from nucypher.crypto.kits import RevocationKit
+        from nucypher.policy.models import Revocation
+
+        revocation = Revocation.from_bytes(request.body)
+        try:
+            with ThreadedSession(self.db_engine) as session:
+                # Verify the Notice was signed by Alice
+                policy_arrangement = self.datastore.get_policy_arrangement(
+                    id_as_hex.encode(), session=session)
+                alice_pubkey = UmbralPublicKey.from_bytes(
+                    policy_arrangement.alice_pubkey_sig.key_data)
+
+                # Check that the request is the same for the provided revocation
+                if id_as_hex != revocation.arrangement_id.hex():
+                    return Response(status_code=400)
+                elif revocation.verify_signature(alice_pubkey):
+                    self.datastore.del_policy_arrangement(
+                        id_as_hex.encode(), session=session)
+        except (NotFound, InvalidSignature):
+            return Response(content='KFrag not found or revocation signature is invalid.', status_code=404)
```

Comment:
... in which case this singular language? (this is effectively a semantic squash, unsure if this is intentional)

Reply:
Completely intentional. It prevents someone from knowing whether the policy is revoked or if the node actually has the KFrag requested if they get an old revocation notice.

Reply:
What exact situation are you talking about? That is why we have the possibility of `failed_revocations` which we can attempt to revoke later.

There is only ever one revocation to one arrangement. Are you saying "some" as in the whole of the revocation process, or am I missing something?

Reply:
Ah - got it.  Yeah - so this is the wrong line for this comment.

>  Are you saying "some" as in the whole of the revocation process

Yes